_posts/2021-04-07-upgrade-improvements.md

```diff
@@ -0,0 +1,66 @@
+And then repeat the same steps for 0.20.
+This made the upgrades hard to execute fully automatically. For example, when installing Strimzi from the [OperatorHub](https://operatorhub.io/operator/strimzi-kafka-operator).
+The rolling update of Kafka cluster was also time-consuming, as upgrading through several versions can take a long time.
+
+#### Process changes
+
+Now, the process is a bit more friendly.
+The main improvement is you don't need to upgrade across all Strimzi versions, For instance, you can upgrade from 0.24 directly to latest without upgrade through any other release like 0.25.
+The upgrade process is shown in the diagram below.
+Strimzi now does the following things regarding Kafka configuration:
+* Detect used Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` (from CR or compute it from Kafka `version`).
+* Configure Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` if it's not specified by user.
+
+Here's how Strimzi handles certain situations:
+* Supported Kafka `version` and `log.message.format.version` or `inter.broker.protocol.version` is set in CR - Strimzi will keep the values and missing one will se based on `version`.
+* Supported Kafka `version` is set, `log.message.format.version` and `inter.broker.protocol.version` are not set in CR - Strimzi will set `log.message.format.version` and `inter.broker.protocol.version` based on `version`
+* Supported Kafka `version` is not set in CR - Strimzi will set Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` based on the default Kafka supported in 0.24.
+* Unsupported Kafka `version` is set in CR - Strimzi will set `Kafka` CR status to `NotReady`. 
+  In that case user has to update CR and manually set a proper config.
+
+You don't need to configure the Kafka `version`, `log.message.format.version` or `inter.broker.protocol.version` before the upgrade, but we **strongly** recommend it.
+
+#### Upgrade without versions
+
+Strimzi is able to deal with upgrade of Kafka clusters where versions are not set in the custom resource.
+In cases like this, Strimzi determines the versions and performs the upgrade as follows:
+* Get the version of the currently deployed Kafka and set it in the custom resource.
+* Perform a rolling update of the Kafka cluster to change the images from the old Strimzi version to the new one.
+* Upgrade the Kafka `version` to the default version supported by the new Strimzi version, which will perform a rolling update.
+* Get `log.message.format.version` and `inter.broker.protocol.version` based on `version`.
+* Set `inter.broker.protocol.version`, and perform a rolling update of the Kafka cluster.
+* Set `log.message.format.version`, and perform a rolling update of the Kafka cluster.
+
+You can see the states of the upgrade process and each rolling update in the diagram below.
+
+![Kafka upgrade states](/assets/images/posts/2021-04-19-kafka-rolling-updates.png)
```

Comment:
I think we should not sue the `latest` version here. Maybe we can use something like _version X_ and _version X+1_ at least for the operator.